src/app/css-variables.ts

```diff
@@ -1,56 +1,74 @@
 import { settingsSelector } from 'app/dim-api/selectors';
+import { deepEqual } from 'fast-equals';
 import { isPhonePortraitSelector } from './shell/selectors';
-import { observeStore } from './utils/redux';
+import { StoreObserver } from './store/observeMiddleware';
 
 function setCSSVariable(property: string, value: { toString: () => string }) {
   if (value) {
     document.querySelector('html')!.style.setProperty(property, value.toString());
   }
 }
 
+export function createItemSizeObserver(): StoreObserver<number> {
+  return {
+    id: 'item-size-observer',
+    getObserved: (rs) => settingsSelector(rs).itemSize,
+    sideEffect: ({ previous, current }) => {
+      if (previous !== current) {
+        setCSSVariable('--item-size', `${Math.max(48, current)}px`);
+      }
+    },
+  };
+}
+
+export function createThemeObserver(): StoreObserver<{ theme: string; isPhonePortrait: boolean }> {
+  return {
+    id: 'theme-observer',
+    equals: deepEqual,
+    getObserved: (rs) => ({
+      theme: settingsSelector(rs).theme,
+      isPhonePortrait: isPhonePortraitSelector(rs),
+    }),
+    sideEffect: ({ previous, current }) => {
+      // TODO (ryan): Hoist the feature flag check up to where the observer is created
+      // Set a class on the body to control the theme. This must be applied on the body for syncThemeColor to work.
+      if ($featureFlags.themePicker && previous.theme !== current.theme) {
+        const themeClass = `theme-${current.theme}`;
+        document.body.className = themeClass;
+        syncThemeColor(current.isPhonePortrait);
+      }
+    },
+  };
+}
+
+export function createTilesPerCharColumnObserver(): StoreObserver<{
+  charCol: number;
+  charColMobile: number;
+  isPhonePortrait: boolean;
+}> {
+  return {
+    id: 'tiles-per-char-column-observer',
+    equals: deepEqual,
+    getObserved: (rs) => ({
+      charCol: settingsSelector(rs).charCol,
+      charColMobile: settingsSelector(rs).charColMobile,
+      isPhonePortrait: isPhonePortraitSelector(rs),
+    }),
+    sideEffect: ({ current }) => {
+      setCSSVariable(
+        '--tiles-per-char-column',
+        current.isPhonePortrait ? current.charColMobile : current.charCol,
+      );
+      syncThemeColor(current.isPhonePortrait);
+    },
+  };
+}
```

Comment:
I decided to radically simplify this as I don't think the complexity that was there was worth the value it added. I created bug while doing this as I accidentally deleted a negation in an `isPhonePortrait` check.

There might be a concern about perf considering this will be slightly noisier on updating `--tiles-per-char-column` but that noise should only be for cases where the current value is the same as the previous value. Unless I am missing something, I don't think that is actually an issue?